arangod/Aql/ExecutionBlockImpl.tpp

```diff
@@ -2582,16 +2614,42 @@ bool ExecutionBlockImpl<Executor>::PrefetchTask::rearmForNextCall(
 
 template<class Executor>
 void ExecutionBlockImpl<Executor>::PrefetchTask::waitFor() const noexcept {
+  uint64_t count = _numberWaiters.fetch_add(1, std::memory_order_relaxed);
+  if (count > 0) {
+    LOG_TOPIC("62515", WARN, Logger::AQL)
+        << "ALERT: Detected " << count + 1
+        << " waiters for a PrefetchTask, stacktrace:";
+    CrashHandler::logBacktrace();
+    _logStacktrace.store(true, std::memory_order_relaxed);
+  }
   std::unique_lock<std::mutex> guard(_lock);
   // (1) - this acquire-load synchronizes with the release-store (3)
-  if (_state.load(std::memory_order_acquire).status == Status::Finished) {
-    return;
+  while (_state.load(std::memory_order_acquire).status != Status::Finished) {
+    std::cv_status s = _bell.wait_for(guard, std::chrono::milliseconds(1000));
+    if (s == std::cv_status::timeout) {
+      auto state = _state.load(std::memory_order_relaxed);
+      // We only log this if the status is not "InProgress", since
+      // this is the only one we expect when a timeout occurs!
+      if (state.status != Status::InProgress) {
+        LOG_TOPIC("62514", WARN, Logger::AQL)
+            << "ALERT: Have waited for a second on an async prefetch task, "
```

Comment:
As discussed, it's generally valid if we see `Status::Finished` here instead. Therefore, in that case, I would print this with LogLevel INFO at most, and without the `ALERT` prefix.

Reply:
Done.